### PR TITLE
Fixes noslip shoes not showing up in nuke ops uplink

### LINKED ISSUE #35602: Nuke ops are unable to purchase noslips
Round ID: 83504

1) Have nuke op uplink.
2) Go to Stealth section.
3) Noslips are missing.

They SHOULD be purchasable, I can see their entry in the uplink_items.dm but it's not showing.

A workaround is that apparently the shoes in the Broken Camokit bundle are noslip.

### PR DESCRIPTION
**Pull request by YPOQ**
Fixes #35602

Basically this allows uplink_items with the same name to exist so long as only one ends up on a given uplink (i.e. they have different gamemode requirements).

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]
If you changed the layout of the list you need to fix the usage of it across the codebase

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -1,45 +1,31 @@
...
-			if(I.exclude_modes.len)
-				if(!gamemode && SSticker && SSticker.mode && (SSticker.mode.type in I.exclude_modes))
-					continue
-				if(gamemode && (gamemode in I.exclude_modes))
-					continue
-			if(I.player_minimum && I.player_minimum > GLOB.joined_player_list.len)
+			if(gamemode && !(gamemode in I.include_modes))
 				continue
+		if(I.exclude_modes.len)
+			if(!gamemode && SSticker && SSticker.mode && (SSticker.mode.type in I.exclude_modes))
```
Comment (Cyberboss) [inline code/modules/uplink/uplink_items.dm]: SSticker always exists

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1,45 +1,31 @@
...
-	for(var/category in GLOB.uplink_items)
-		for(var/item in GLOB.uplink_items[category])
-			var/datum/uplink_item/I = GLOB.uplink_items[category][item]
-			if(!istype(I))
+	for(var/path in GLOB.uplink_items)
+		var/datum/uplink_item/I = new path
+		if(!I.item)
+			continue
+		if(I.include_modes.len)
+			if(!gamemode && SSticker && SSticker.mode && !(SSticker.mode.type in I.include_modes))
```
Comment (Cyberboss) [inline code/modules/uplink/uplink_items.dm]: SSticker always exists

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -110,11 +96,6 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 	to_chat(user, "[A] materializes onto the floor.")
 	return A
 
-/datum/uplink_item/Destroy()
-	if(src in GLOB.uplink_items)
-		GLOB.uplink_items -= src	//Take us out instead of leaving a null!
```
Comment (Cyberboss) [inline code/modules/uplink/uplink_items.dm]: This should still exist in some form

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/uplink/uplink_items.dm (hunk 3 above)
Comment (YPOQ) [reply code/modules/uplink/uplink_items.dm]: uplink_items is a list of paths now

**Pull request merged**